src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -90,7 +100,7 @@ AppHost::AppHost() noexcept :
 AppHost::~AppHost()
 {
     // destruction order is important for proper teardown here
-
+    _DestroyTrayIcon();

```

Comment:
(fair warning, I may be completely wrong in how I'm thinking about this)
We have one `AppHost` per window right? So why are we destroying the tray icon each time we close a window? Shouldn't we just be updating the tray with which windows there are? Isn't it the monarch's responsibility to actually destroy the tray icon if there are no more windows?

Reply:
Yup, there's only one `AppHost` per window, and every Tray Icon knows what window it's associated with by `HWND`. But, I only want the Monarch to be able to create a Tray Icon and thus only the Monarch should be able to destroy a tray icon because every other `AppHost` won't have one. `_DestroyTrayIcon` checks if we have a `_trayIconData` which a Monarch would only have, but I should be more explicit about actually checking `IsMonarch` for understandability.

So since a Monarch's `AppHost` should be the only one that has a Tray Icon, if it ever dies, we'll need to destroy the icon it's associated with and then a new `AppHost`/Window will become Monarch and they will now have to create their own as I can't just hand over the old tray icon data.